src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java

```
@@ -3,126 +3,96 @@
+    // and return the string "null" if there is missing data for an optional field
 
-        public String getColumnValue(final VariantContext variant, ProbeInfo probeInfo, String sampleId) {
-            String normx = NORMX.getColumnValue(variant, probeInfo, sampleId);
-            String normy = NORMY.getColumnValue(variant, probeInfo, sampleId);
-            String baf = BAF.getColumnValue(variant, probeInfo, sampleId);
-            String lrr = LRR.getColumnValue(variant, probeInfo, sampleId);
+    sample_id,
+    probe_id,
+    GT_encoded { // Required
+        public String getColumnValue(final VariantContext variant) {
+            List<Integer> alleleIndexes = CommonCode.getGTAlleleIndexes(variant);
 
-            RawArrayData d = new RawArrayData(convert(normx),
-                                              convert(normy),
-                                              convert(baf),
-                                              convert(lrr)
-                                              );
-            return String.valueOf(d.encode());
-        }
-    },
-
-    probe_id { // Required
-        public String getColumnValue(final VariantContext variant, ProbeInfo probeInfo, String sampleId) {
-            return String.valueOf(probeInfo.probeId);
-        }
-    },
-
-    GT_encoded {
-        public String getColumnValue(final VariantContext variant, ProbeInfo probeInfo, String sampleId) {
-            Genotype g = variant.getGenotype(0);
             RawArrayTsvCreator.GT_encoding gt = RawArrayTsvCreator.GT_encoding.MISSING;
-            if (g.isHomRef() || g.isHomVar()) {
-                Allele allele = g.getAllele(0);
-                if (allele.basesMatch(probeInfo.alleleA)) {
-                    gt = RawArrayTsvCreator.GT_encoding.AA;
-                } else if (allele.basesMatch(probeInfo.alleleB)) {
-                    gt = RawArrayTsvCreator.GT_encoding.BB;
+            if (alleleIndexes.size() == 2) {
```

what if it's not two?  should we throw an exception?

---

added a warning